sync-api-common/src/browser/connection.ts

```diff
@@ -4,42 +4,62 @@
  * ------------------------------------------------------------------------------------------ */
 
 import RAL from '../common/ral';
-import { BaseServiceConnection, BaseClientConnection, Message, RequestType } from '../common/connection';
+import { BaseServiceConnection, BaseClientConnection, Message, RequestType, KnownConnectionIds, BroadcastChannelName } from '../common/connection';
 
 export class ClientConnection<Requests extends RequestType | undefined = undefined> extends BaseClientConnection<Requests> {
 
-	private readonly port: MessagePort | Worker | DedicatedWorkerGlobalScope;
+	private readonly channel: BroadcastChannel;
 
-	constructor(port: MessagePort | Worker | DedicatedWorkerGlobalScope) {
-		super();
-		this.port = port;
-		this.port.onmessage = ((event: MessageEvent<Message>) => {
-			this.handleMessage(event.data);
-		});
+	constructor(channelName: string = BroadcastChannelName) {
+		super(self.location.pathname);
+		this.channel = new BroadcastChannel(channelName);
+		this.channel.addEventListener('message', this._handleMessageEvent.bind(this));
+	}
+
+	dispose() {
+		this.channel.removeEventListener('message', this._handleMessageEvent.bind(this));
```

Comment:
Not happy with requiring a dispose, but since Broadcast channels are global, they need to be cleaned up (don't go away on Worker shutdown)

An alternative would be to force a new channel name on each Client/Server pair, but that would be hard to reconcile because the Client is created in the worker and doesn't have access to the memory of the Server.

Additionally, the BroadcastChannel only works on the same origin. Would the same origin be used for the extension host as the workers in vscode.dev?